packages/loader/container-loader/src/contracts.ts

```diff
@@ -167,6 +167,11 @@ export interface IConnectionManagerFactoryArgs {
 	 * `undefined` indicates that user permissions are not yet known.
 	 */
 	readonly readonlyChangeHandler: (readonly?: boolean) => void;
+
+	/**
+	 * Called whenever we try to start establishing a new connection.
+	 */
```

Comment:
might be worth adding that this only happens in specific cases like container boot, container.connect(). I.e. it's not invoked when reconnecting for any reason.